csrc/logical_domain_map.cpp

```diff
@@ -325,6 +325,27 @@ std::unordered_map<IterDomain*, IterDomain*> PairwiseLogicalDomainMap::map(
     return dom_map;
   }
 
+  if (IndexAccumulateOp* iaop =
```

Comment:
You're not the first, but I'd really appreciate if you could try to avoid expanding this function with yet another op-specific conditional block. This function used to be a relatively small function, but now it's hundreds of lines of code, making it difficult to oversee the whole function. This is one of the most important functions in nvFuser as it's the basis of many things, including IdModel.